### PR TITLE
fix kotlin warnings in CDK core

### PR DESCRIPTION
**Pull request by stephane-airbyte**
just fixing kotlin warnings in CDK

Interesting tidbit: Because of named parameters, it's important that a function override has the same parameter names as the overriden function. Hence some parameter renaming here.

Also, variable name shadowing is reported as a warning

### CODE REVIEW

**Comment by stephane-airbyte** [comment]
* **#37477** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/37477?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @stephane-airbyte and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/db/jdbc/AbstractJdbcCompatibleSourceOperations.kt [hunk 1]
```diff
@@ -332,8 +332,10 @@ abstract class AbstractJdbcCompatibleSourceOperations<Datatype> :
         }
     }
 
+    // Note: I'm guessing this is an open function, since it doesn't do anything here, and doesn't
+    // override anything either
```
Comment (postamar) [inline airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/db/jdbc/AbstractJdbcCompatibleSourceOperations.kt:336]: I can answer that: they're all open.

File: airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/db/jdbc/JdbcSourceOperations.kt [hunk 2]
```diff
@@ -17,7 +17,7 @@ import org.slf4j.LoggerFactory
 /** Implementation of source operations with standard JDBC types. */
 open class JdbcSourceOperations :
     AbstractJdbcCompatibleSourceOperations<JDBCType>(), SourceOperations<ResultSet, JDBCType> {
-    protected fun safeGetJdbcType(columnTypeInt: Int): JDBCType {
+    private fun safeGetJdbcType(columnTypeInt: Int): JDBCType {
```
Comment (postamar) [inline airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/db/jdbc/JdbcSourceOperations.kt:20]: This probably needs to be open protected as well, but I'm less sure.

File: airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/base/IntegrationRunner.kt [hunk 3]
```diff
@@ -256,7 +251,7 @@ internal constructor(
         messageIterator: AutoCloseableIterator<AirbyteMessage>,
         recordCollector: Consumer<AirbyteMessage>
     ) {
-        messageIterator!!.airbyteStream.ifPresent { s: AirbyteStreamNameNamespacePair? ->
+        messageIterator.airbyteStream.ifPresent { s: AirbyteStreamNameNamespacePair? ->
```
Comment (postamar) [inline airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/base/IntegrationRunner.kt:254]: s need not be nullable. Comment applies elsewhere also.

**Review by postamar — APPROVED** [review]

File: airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/db/jdbc/AbstractJdbcCompatibleSourceOperations.kt (hunk 1 above)
Comment (stephane-airbyte) [reply airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/db/jdbc/AbstractJdbcCompatibleSourceOperations.kt:336]: I think they should be abstract. I really don't like optional methods. Battle for another day though

File: airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/base/IntegrationRunner.kt (hunk 3 above)
Comment (stephane-airbyte) [reply airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/base/IntegrationRunner.kt:254]: will need another pass

**Pull request merged**